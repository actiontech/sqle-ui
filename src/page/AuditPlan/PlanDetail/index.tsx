import { useRequest } from 'ahooks';
import { PageHeader, Typography } from 'antd';
import { useTranslation } from 'react-i18next';
import { Redirect, Route, Switch, useParams } from 'react-router-dom';
import audit_plan from '../../../api/audit_plan';
import BackButton from '../../../components/BackButton';
import { useCurrentProjectName } from '../../ProjectManage/ProjectDetail';
import PlanDetail from './Detail';
import { PlanDetailUrlParams } from './index.type';

const PlanDetailPage = () => {
  const urlParams = useParams<PlanDetailUrlParams>();
  const { projectName } = useCurrentProjectName();
  const { t } = useTranslation();
  const { data: auditTask } = useRequest(
    () => {
      return audit_plan.getAuditPlanV1({
        audit_plan_name: urlParams.auditPlanName,
        project_name: projectName,
      });
    },
    {
      formatResult(res) {
        return res.data;
      },
    }
  );

  return (
    <>
      <PageHeader
        ghost={false}
        title={t('auditPlan.detailPage.pageTitle', {
          name: urlParams.auditPlanName,
        })}
        extra={[<BackButton key="goBack" />]}
      >
        <Typography.Paragraph>
          {t('auditPlan.detailPage.auditTaskType', {
            type:
              auditTask?.data?.audit_plan_meta?.audit_plan_type_desc ?? '--',
          })}
        </Typography.Paragraph>
        {t('auditPlan.detailPage.pageDesc')}
      </PageHeader>
      <section className="padding-content">
        <Switch>
          <Route
            path="/project/:projectName/auditPlan/detail/:auditPlanName"
            component={PlanDetail}
          />
          <Redirect to="/project/:projectName/auditPlan" />
        </Switch>
      </section>
    </>
  );
};

export default PlanDetailPage;
