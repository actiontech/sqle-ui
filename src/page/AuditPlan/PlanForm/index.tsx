import { useBoolean } from 'ahooks';
import { Button, Form, Input, Select, Space } from 'antd';
import { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { IAuditPlanParamResV1 } from '../../../api/common';
import useAsyncParams from '../../../components/BackendForm/useAsyncParams';
import CronInput from '../../../components/CronInput';
import { PageBigFormLayout } from '../../../data/common';
import EmitterKey from '../../../data/EmitterKey';
import { checkCron } from '../../../hooks/useCron/cron.tool';
import useGlobalRuleTemplate from '../../../hooks/useGlobalRuleTemplate';
import useRuleTemplate from '../../../hooks/useRuleTemplate';
import EventEmitter from '../../../utils/EventEmitter';
import { nameRule } from '../../../utils/FormRule';
import { AuditTaskType } from './AuditTaskType';
import { DataSource, DataSourceProps } from './DataSource';
import { PlanFormField, PlanFormProps } from './index.type';

const PlanForm: React.FC<PlanFormProps> = ({ form, ...props }) => {
  const { t } = useTranslation();

  const [dbType, setDbType] = useState('');
  const [dataSource, setDataSource] = useState('');

  const [submitLoading, { setTrue: startSubmit, setFalse: submitFinish }] =
    useBoolean();

  const [asyncParams, setAsyncParams] = useState<
    IAuditPlanParamResV1[] | undefined
  >([]);

  const { mergeFromValueIntoParams } = useAsyncParams();
  const { globalRuleTemplateList, updateGlobalRuleTemplateList } =
    useGlobalRuleTemplate();

  const { ruleTemplateList, updateRuleTemplateList } = useRuleTemplate();

  const submit = (values: PlanFormField) => {
    if (values.params && asyncParams) {
      const params = values.params;
      delete values.params;
      values.asyncParams = mergeFromValueIntoParams(params, asyncParams);
    }
    startSubmit();
    props.submit(values).finally(() => {
      submitFinish();
    });
  };

  const resetForm = () => {
    setDataSource('');
    setDbType('');
    EventEmitter.emit(EmitterKey.Reset_Audit_Plan_Form_Instance_List);
    if (!!props.defaultValue) {
      form.resetFields(['databaseName', 'cron', 'schema']);
    } else {
      form.resetFields();
    }
  };

  const dbTypeChange: DataSourceProps['dbTypeChange'] = (dbType) => {
    setDbType(dbType);
    form.setFieldsValue({
      ruleTemplateName: undefined,
    });
  };

  useEffect(() => {
    if (props.defaultValue) {
      form.setFieldsValue({
        name: props.defaultValue.audit_plan_name,
        databaseName: props.defaultValue.audit_plan_instance_name,
        schema: props.defaultValue.audit_plan_instance_database,
        cron: props.defaultValue.audit_plan_cron,
        dbType: props.defaultValue.audit_plan_db_type,
        ruleTemplateName: props.defaultValue.rule_template_name,
      });
      if (!!props.defaultValue.audit_plan_instance_name) {
        setDataSource(props.defaultValue.audit_plan_instance_name);
      }
      if (!!props.defaultValue.audit_plan_db_type) {
        setDbType(props.defaultValue.audit_plan_db_type);
      }
    }
    updateRuleTemplateList(props.projectName);
    updateGlobalRuleTemplateList();
  }, [
    form,
    props.defaultValue,
    props.projectName,
    updateGlobalRuleTemplateList,
    updateRuleTemplateList,
  ]);

  useEffect(() => {
    const reset = () => {
      resetForm();
    };
    EventEmitter.subscribe(EmitterKey.Rest_Audit_Plan_Form, reset);
    return () => {
      EventEmitter.unsubscribe(EmitterKey.Rest_Audit_Plan_Form, reset);
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <Form {...PageBigFormLayout} form={form} onFinish={submit}>
      <Form.Item
        label={t('auditPlan.planForm.name')}
        name="name"
        rules={[
          {
            required: true,
          },
          ...nameRule(),
        ]}
      >
        <Input
          disabled={!!props.defaultValue}
          placeholder={t('common.form.placeholder.input')}
        />
      </Form.Item>
      <DataSource
        dataSource={dataSource}
        form={form}
        dataSourceChange={setDataSource}
        dbTypeChange={dbTypeChange}
        defaultValue={props.defaultValue}
        projectName={props.projectName}
      />
      <AuditTaskType
        dbType={dbType}
        form={form}
        updateCurrentTypeParams={setAsyncParams}
        defaultValue={props.defaultValue}
      />

      <Form.Item
        hidden={!dbType}
        label={t('auditPlan.planForm.ruleTemplateName')}
        name="ruleTemplateName"
        tooltip={t('auditPlan.planForm.ruleTemplateNameTips')}
      >
        <Select placeholder={t('common.form.placeholder.select')}>
          {[...ruleTemplateList, ...globalRuleTemplateList].map((template) => {
            if (template.db_type === dbType) {
              return (
                <Select.Option
                  key={template.rule_template_name}
                  value={template.rule_template_name ?? ''}
                >
                  {template.rule_template_name}
                </Select.Option>
              );
            }
            return null;
          })}
        </Select>
      </Form.Item>

      <Form.Item
        label={t('auditPlan.planForm.cron')}
        name="cron"
        initialValue="0 0 * * *"
        rules={[
          {
            required: true,
          },
          {
            validator(_, value) {
              const error = checkCron(value);
              if (error === '') {
                return Promise.resolve();
              }
              return Promise.reject(t(error));
            },
          },
        ]}
      >
        <CronInput />
      </Form.Item>
      <Form.Item label=" " colon={false}>
        <Space>
          <Button htmlType="submit" type="primary" loading={submitLoading}>
            {t('common.submit')}
          </Button>
          <Button onClick={resetForm} disabled={submitLoading}>
            {t('common.reset')}
          </Button>
        </Space>
      </Form.Item>
    </Form>
  );
};

export default PlanForm;
