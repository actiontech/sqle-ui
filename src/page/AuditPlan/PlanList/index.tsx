import { SyncOutlined } from '@ant-design/icons';
import { useBoolean, useRequest } from 'ahooks';
import { Button, Card, message, Space, Table } from 'antd';
import { useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { useDispatch } from 'react-redux';
import { Link, useHistory } from 'react-router-dom';
import audit_plan from '../../../api/audit_plan';
import { IAuditPlanResV1 } from '../../../api/common';
import { ResponseCode } from '../../../data/common';
import { ModalName } from '../../../data/ModalName';
import useTable from '../../../hooks/useTable';
import {
  initAuditPlanModalStatus,
  updateAuditPlanModalStatus,
  updateSelectAuditPlan,
} from '../../../store/auditPlan';
import PlanListModal from './Modal';
import { planListTableHeader } from './tableColumn';

const PlanList = () => {
  const { t } = useTranslation();
  const history = useHistory();
  const [
    removePending,
    { setTrue: startRemoveAuditPlan, setFalse: removeAuditPlanFinish },
  ] = useBoolean();

  const { pagination, tableChange } = useTable();

  const { data, loading, refresh } = useRequest(
    () => {
      return audit_plan.getAuditPlansV1({
        page_index: pagination.pageIndex,
        page_size: pagination.pageSize,
      });
    },
    {
      paginated: true,
      refreshDeps: [pagination],
      formatResult(res) {
        return {
          total: res.data?.total_nums ?? 1,
          list: res.data?.data ?? [],
        };
      },
    }
  );

  const removeAuditPlan = (auditPlanName: string) => {
    if (removePending) {
      return;
    }
    const hide = message.loading(
      t('auditPlan.remove.loading', { name: auditPlanName }),
      0
    );
    startRemoveAuditPlan();
    audit_plan
      .deleteAuditPlanV1({
        audit_plan_name: auditPlanName,
      })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          message.success(
            t('auditPlan.remove.successTips', { name: auditPlanName })
          );
          refresh();
        }
      })
      .finally(() => {
        hide();
        removeAuditPlanFinish();
      });
  };

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(
      initAuditPlanModalStatus({
        modalStatus: {
          [ModalName.Subscribe_Notice]: true,
        },
      })
    );
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const openModal = (name: ModalName, row?: IAuditPlanResV1) => {
    if (row) {
      dispatch(updateSelectAuditPlan(row));
    }

    dispatch(
      updateAuditPlanModalStatus({
        modalName: name,
        status: true,
      })
    );
  };

  return (
    <Card
      title={
        <Space>
          {t('auditPlan.list.title')}
          <Button onClick={refresh}>
            <SyncOutlined spin={loading} />
          </Button>
        </Space>
      }
      extra={[
        <Link key="create-audit-plan" to="/auditPlan/create">
          <Button type="primary">{t('auditPlan.action.create')}</Button>
        </Link>,
      ]}
    >
      <Table
        columns={planListTableHeader(removeAuditPlan, openModal)}
        dataSource={data?.list ?? []}
        rowKey="audit_plan_name"
        pagination={{
          total: data?.total,
          showSizeChanger: true,
        }}
        loading={loading}
        onRow={(record) => ({
          onClick: () => {
            history.push(`/auditPlan/detail/${record.audit_plan_name}`);
          },
        })}
        onChange={tableChange}
      />
      <PlanListModal />
    </Card>
  );
};

export default PlanList;
