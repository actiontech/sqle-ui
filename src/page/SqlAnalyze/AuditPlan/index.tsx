import { useBoolean } from 'ahooks';
import { ResultStatusType } from 'antd/lib/result';
import { useCallback, useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import audit_plan from '../../../api/audit_plan';
import { ISQLExplain, ITableMeta } from '../../../api/common';
import { ResponseCode } from '../../../data/common';
import SqlAnalyze from '../SqlAnalyze';
import { AuditPlanReportSqlAnalyzeUrlParams } from './index.type';

const AuditPlanSqlAnalyze = () => {
  const urlParams = useParams<AuditPlanReportSqlAnalyzeUrlParams>();

  const [errorMessage, setErrorMessage] = useState<string>('');

  const [sqlExplain, setSqlExplain] = useState<ISQLExplain>();
  const [tableSchemas, setTableSchemas] = useState<ITableMeta[]>([]);
  const [
    loading,
    { setTrue: startGetSqlAnalyze, setFalse: getSqlAnalyzeFinish },
  ] = useBoolean();

  const [errorType, setErrorType] = useState<ResultStatusType>('error');

  const getSqlAnalyze = useCallback(async () => {
    startGetSqlAnalyze();
    try {
      const res = await audit_plan.getTaskAnalysisData({
        audit_plan_report_id: urlParams.reportId,
        number: urlParams.sqlNum,
      });
      if (res.data.code === ResponseCode.SUCCESS) {
        setErrorMessage('');
        setSqlExplain(res.data.data?.sql_explain);
        setTableSchemas(res.data.data?.table_metas ?? []);
      } else {
        if (res.data.code === ResponseCode.NotSupportDML) {
          setErrorType('info');
        } else {
          setErrorType('error');
        }
        setErrorMessage(res.data.message ?? '');
      }
    } finally {
      getSqlAnalyzeFinish();
    }
  }, [
    getSqlAnalyzeFinish,
    startGetSqlAnalyze,
    urlParams.sqlNum,
    urlParams.reportId,
  ]);

  useEffect(() => {
    getSqlAnalyze();
  }, [getSqlAnalyze]);

  return (
    <SqlAnalyze
      tableSchemas={tableSchemas}
      sqlExplain={sqlExplain}
      errorType={errorType}
      errorMessage={errorMessage}
      loading={loading}
    />
  );
};

export default AuditPlanSqlAnalyze;
