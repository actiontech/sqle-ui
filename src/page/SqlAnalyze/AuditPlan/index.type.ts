export type AuditPlanReportSqlAnalyzeUrlParams = {
  reportId: string;
  sqlNum: string;
  auditPlanName: string;
};
