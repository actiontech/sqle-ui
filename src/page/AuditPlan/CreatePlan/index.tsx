import { useBoolean } from 'ahooks';
import { Button, Card, Modal, Result } from 'antd';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import audit_plan from '../../../api/audit_plan';
import BackButton from '../../../components/BackButton';
import { ResponseCode } from '../../../data/common';
import EmitterKey from '../../../data/EmitterKey';
import EventEmitter from '../../../utils/EventEmitter';
import { useCurrentProjectName } from '../../ProjectManage/ProjectDetail';
import PlanForm from '../PlanForm';
import { PlanFormField } from '../PlanForm/index.type';

const CreateAuditPlan = () => {
  const { t } = useTranslation();
  const [visible, { setTrue: openResultModal, setFalse: closeResultModal }] =
    useBoolean();
  const { projectName } = useCurrentProjectName();

  const createAuditPlan = (values: PlanFormField) => {
    return audit_plan
      .createAuditPlanV1({
        project_name: projectName,
        audit_plan_cron: values.cron,
        audit_plan_instance_database: values.schema,
        audit_plan_instance_name: values.databaseName,
        audit_plan_instance_type: values.dbType,
        audit_plan_name: values.name,
        audit_plan_type: values.auditTaskType,
        audit_plan_params: values.asyncParams,
        rule_template_name: values.ruleTemplateName,
      })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          openResultModal();
        }
      });
  };

  const resetAndCloseResultModal = () => {
    EventEmitter.emit(EmitterKey.Rest_Audit_Plan_Form);
    closeResultModal();
  };

  return (
    <Card
      title={t('auditPlan.create.title')}
      extra={[<BackButton key="goBack" />]}
    >
      <PlanForm submit={createAuditPlan} />
      <Modal
        title={t('common.operateSuccess')}
        footer={null}
        closable={false}
        visible={visible}
      >
        <Result
          status="success"
          title={t('auditPlan.create.successTitle')}
          subTitle={
            <Link to={`/projectName/${projectName}/auditPlan`}>
              {t('auditPlan.create.successGuide')} {'>'}
            </Link>
          }
          extra={[
            <Button key="close" onClick={closeResultModal}>
              {t('common.close')}
            </Button>,
            <Button
              type="primary"
              key="resetAndClose"
              onClick={resetAndCloseResultModal}
            >
              {t('common.resetAndClose')}
            </Button>,
          ]}
        />
      </Modal>
    </Card>
  );
};

export default CreateAuditPlan;
