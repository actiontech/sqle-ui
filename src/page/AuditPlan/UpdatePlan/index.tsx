import { useBoolean } from 'ahooks';
import { Card, Modal, Result, Button } from 'antd';
import { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Link, useParams } from 'react-router-dom';
import audit_plan from '../../../api/audit_plan';
import BackButton from '../../../components/BackButton';
import { ResponseCode } from '../../../data/common';
import EmitterKey from '../../../data/EmitterKey';
import EventEmitter from '../../../utils/EventEmitter';
import PlanForm from '../PlanForm';
import { PlanFormField } from '../PlanForm/index.type';
import { UpdateAuditPlanUrlParams } from './index.type';

const UpdateAuditPlan = () => {
  const { t } = useTranslation();
  const urlParams = useParams<UpdateAuditPlanUrlParams>();
  const [visible, { setTrue: openResultModal, setFalse: closeResultModal }] =
    useBoolean();
  const [defaultValue, setDefaultValue] = useState<any>();

  const updateAuditPlan = (values: PlanFormField) => {
    return audit_plan
      .updateAuditPlanV1({
        audit_plan_cron: values.cron,
        audit_plan_instance_database: values.schema,
        audit_plan_instance_name: values.databaseName,
        audit_plan_name: urlParams.auditPlanName,
      })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          openResultModal();
        }
      });
  };

  const updateAndCloseResultModal = () => {
    EventEmitter.emit(EmitterKey.Rest_Audit_Plan_Form);
    closeResultModal();
    getCurrentAuditPlan();
  };

  const getCurrentAuditPlan = () => {
    setTimeout(() => {
      setDefaultValue({
        audit_plan_cron: '0 */2 * * *',
        audit_plan_db_type: 'mysql',
        audit_plan_instance_database: 'sqle',
        audit_plan_name: 'db1',
        audit_plan_instance_name: 'db1',
      });
    }, 1000);
  };

  useEffect(() => {
    getCurrentAuditPlan();
  }, []);

  return (
    <Card
      title={t('auditPlan.update.title')}
      extra={[<BackButton key="goBack" />]}
    >
      <PlanForm submit={updateAuditPlan} defaultValue={defaultValue} />
      <Modal
        title={t('common.operateSuccess')}
        footer={null}
        closable={false}
        visible={visible}
      >
        <Result
          status="success"
          title={t('auditPlan.update.successTitle')}
          subTitle={
            <Link to="/auditPlan">
              {t('auditPlan.update.successGuide')} {'>'}
            </Link>
          }
          extra={[
            <Button key="close" onClick={updateAndCloseResultModal}>
              {t('common.close')}
            </Button>,
          ]}
        />
      </Modal>
    </Card>
  );
};

export default UpdateAuditPlan;
