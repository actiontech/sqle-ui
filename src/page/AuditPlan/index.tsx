import { PageHeader } from 'antd';
import { useTranslation } from 'react-i18next';
import { Redirect, Route, Switch } from 'react-router-dom';
import CreateAuditPlan from './CreatePlan';
import PlanList from './PlanList';

const AuditPlan = () => {
  const { t } = useTranslation();

  return (
    <>
      <PageHeader ghost={false} title={t('auditPlan.pageTitle')}>
        {t('auditPlan.pageDesc')}
      </PageHeader>

      <section className="padding-content">
        <Switch>
          <Route path="/auditPlan" exact={true} component={PlanList} />
          <Route path="/auditPlan/create" component={CreateAuditPlan} />
          <Redirect to="/auditPlan" />
        </Switch>
      </section>
    </>
  );
};

export default AuditPlan;
