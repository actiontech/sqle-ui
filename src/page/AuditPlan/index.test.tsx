import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import AuditPlan from '.';

describe('AuditPlan', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<AuditPlan />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
