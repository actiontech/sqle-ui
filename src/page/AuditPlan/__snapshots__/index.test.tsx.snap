// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuditPlan should match snapshot 1`] = `
<Fragment>
  <PageHeader
    ghost={false}
    title="auditPlan.pageTitle"
  >
    auditPlan.pageDesc
  </PageHeader>
  <section
    className="padding-content"
  >
    <Outlet />
  </section>
</Fragment>
`;
